lib/service/service.go

```diff
@@ -2065,6 +2067,26 @@ func (process *TeleportProcess) initSSH() error {
 	return nil
 }
 
+func compareVersions(c *Connector) error {
+	sver, err := c.Client.GetVersion(context.TODO())
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
The server auth version is already present during `registerWithAutherServer`  call in  `registerWithAuthServer->reconnectToAuthService` `->` `pingResponse.ServerVersion`
https://github.com/gravitational/teleport/blob/master/lib/service/connect.go#L55
thus there is no need to introduce and invoke additional `GetVersion` call.